src/graphql.js

```diff
@@ -628,4 +628,258 @@ function authAPIcall(query,variables,callback,cacheKey,timeFresh,useLocalStorage
 	APIcallsUsed++
 }
 const ANILIST_QUERY_LIMIT = 90;
+
+localforage.config({name: "automail"});
+
+//begin api v2
+const apiCache = localforage.createInstance({name: "automail", storeName: "api"});
+let apiResetLimit;
+
+/** Provides default arguments for an {@link anilistAPI()} request */
+class QueryOptions{
+	/**
+	 * Creates an arguments object
+	 * @param {object} [args={}] - Contains the options which will override the default options for the request.
+	 */
+	constructor(args={}){
+		this.variables = {};
+		this.cacheKey = null;
+		this.duration = null;
+		this.overwrite = false;
+		this.auth = false;
+		this.internal = false;
+		Object.assign(this, args);
+	}
+}
+
+/** Configures options for a network request */
+class RequestOptions{
+	constructor(query, variables, auth, internal){
+		this.method = "POST",
+		this.headers = new Headers({
+			"Content-Type": "application/json",
+			"Accept": "application/json"
+		}),
+		this.body = JSON.stringify({
+			"query": query,
+			"variables": variables
+		})
+		this.#isAuth(auth)
+		this.#isInternal(internal)
+	}
+	#isAuth(auth){
+		if(auth === true && useScripts.accessToken){
+			this.headers.set("Authorization", "Bearer " + useScripts.accessToken)
+		}
+	}
+	#isInternal(internal){
+		if(internal === true){
+			if(al_token){
+				this.headers.set("x-csrf-token", al_token)
+			}
+			this.headers.set("schema", "internal")
+		}
+	}
+}
+
+/**
+ * Updates the variables tracking the API request limit
+ * @param {Response} res - The Response object from a network request.
+ */
+function updateLimit(res){
+	if(res.headers.has("x-ratelimit-limit")){
+		APIlimit = res.headers.get("x-ratelimit-limit");
+	}
+	if(res.headers.has("x-ratelimit-remaining")){
+		APIcallsUsed = APIlimit - res.headers.get("x-ratelimit-remaining");
+	}
+}
+
+/**
+ * Checks the API cache for an existing item and returns it if it has not expired
+ * @param {string} key - The key to check for in the datastore.
+ * @returns {Promise<object>} The cached data if found.
+ */
+async function checkCache(key){
+	const item = await apiCache.getItem(key);
+	if(item){
+		if(!item.expiresAt || (NOW() < item.expiresAt)){
+			return item.data;
+		}
+		else{
+			return apiCache.removeItem(key);
+		}
+	}
+	return;
+}
+
+/**
+ * Iterates the API cache for expired items and removes them
+ * @returns {Promise}
+ */
+async function flushCache(){
+	return apiCache.iterate((item, key) => {
+		if(NOW() > item.expiresAt){
+			apiCache.removeItem(key);
+		}
+	})
+}
+
+/**
+ * Saves data to the API cache
+ * @param {string} key - The key used to store the data.
+ * @param {object} data - The data to store.
+ * @param {number} [duration] - The length of time to store in seconds.
```

Comment:
Why change the API to seconds instead of milliseconds? My reasoning for picking ``ms`` was that when you write ``*1000``, there's no ambiguity.

Reply:
I thought it would simplify things since I didn't think that kind of granularity was needed. I'm not against changing it back though.

Reply:
If there's no strong reason for the change, I prefer keeping it the same as in the v1 api to make conversion of modules less risky.

Reply:
ok i will change it back soon